### ISSUE TITLE
CDP Runtime.evaluate does not drain async tasks 

### ISSUE THREAD

**Issue by EmielM** [facebook/hermes#1531]
## Bug Description

I'm trying to run some automated testing scripts through the Hermes CDP interface using [Runtime.evaluate](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#method-evaluate).

First off, the `awaitPromises` flag does not seem to be supported at all. There are some work-arounds to this, so that's fine.

A bigger issue seems to be that after `Runtime.evaluate`, planned Promise tasks are not executed. What I see is usually after a second or 5-10, the queue is triggered and drained again. I then tried to workaround this by running `HermesInternal.drainJobs` after every command, but [this is unexposed](https://github.com/facebook/hermes/blob/add4115869df5b78914fab0ca15d71ab958e8588/lib/VM/JSLib/HermesInternal.cpp#L815) without hermes-internal-test-methods flag see; I didn't manage to enable that flag in an otherwise vanilla RN project either.

Tried to create a simple case to replicate, but the overall setup is pretty involving.

**Comment by EmielM** [comment]
Apologies, I c/should have been a bit more detailed, indeed. Thanks for your patience.

I attach a chrome inspector via CDP, then evaluate `(new Promise((resolve) => resolve('hey'))).then((x) => console.log(x));`.

Here's a screenshot that illustrates the (timing) issue:
<img width="528" alt="hermes-lagging-promise-queue" src="https://github.com/user-attachments/assets/5db4344c-7016-4326-a5e1-5ab8ecf63178">

My hypothesis is that the task queue is simply not drained in the Hermes CDP client, and after a while some other message goes over the RN bridge or so and that triggers the pending queue to flush.

Logging the backend CDP interaction:
```
> {"id":552,"method":"Runtime.evaluate","params":{"expression":"(new Promise((resolve) =\u003e resolve('hey'))).then((x) =\u003e console.log(x
));","objectGroup":"console","includeCommandLineAPI":true,"silent":false,"returnByValue":false,"generatePreview":true,"userGesture":true,"awaitPromise":false,"replMode":true,"allowUnsafeEvalBlockedByCSP":false,"contextId":1}}
< {"id":552,"result":{"result":{"className":"Object","description":"Object","objectId":"453","preview":{"description":"Object","overflow":false,"properties":[{"name":"_h","type":"number","value":"0"},{"name":"_i","type":"number","value":"0"},{"name":"_j","subtype":"null","type":"object","value":"null"},{"name":"_k","subtype":"null","type":"object","value":"null"}],"type":"object"},"type":"object"}}}                                                                    
```

One little update: I did find out that `HermesInternal.useEngineQueue()` is `false` in my setup.

Will attempt to recreate in latest RN/Hermes, but that will take some time in my setup unfortunately.

**Comment by tmikov** [comment]
> One little update: I did find out that HermesInternal.useEngineQueue() is false in my set

@EmielM Right. That is only enabled with the New Architecture. But either way, Hermes doesn't control the event loop and the draining of the queue. I think we should transfer this issue to React Native.

**Comment by dannysu** [comment]
@EmielM please note that we also don't support just any Chrome DevTools, so if you're just opening up Chrome browser and using it then things are not 100% compatible. We only support the specific fork of DevTools we're calling React DevTools.

The promise queue draining behavior looks to be RN related, so I'll transfer this issue over.
